### PR TITLE
refactor(lib): convert peer methods to use Dispatch pattern

### LINKED ISSUE #1722: refactor(lib, api): convert peers lib methods to use dispatch, adjust handler methods accordingly
- [ ] List (PeersHandler)
- [ ] ConnectedIPFSPeers (ConnectionsHandler)
- [ ] ConnectedQriProfiles (QriConnectionsHandler)
- [ ] ConnectToPeer (ConnectToPeer)
- [ ] DisconnectFromPeer (ConnectToPeer)
- [ ] Info (PeerHandler)

- added `inst.Peer()` for returning `PeerMethods`
- added `PeerMethod.Name` & `PeerMethod.Attributes`
- adding "/disconnect" & "/disconnect/{path:.**}" endpoints
- renamed some functions for clarity:
    - "ConnectToPeer" -> "Connect" (so it will read inst.Peer().Connect())
    - "DisconnectFromPeer" -> "Disconnect" (so it will read inst.Peer().Disconnect())
    - "ConnectedIPFSPeers" -> "Connections" (so it will read inst.Peer().Connections())
- converted lib methods to dispatch
- added `ConnectionsParams`, but limit and offset either don't do anything or aren't both used, should document in the code both in the params and at the lib function

to do:
- remove NewPeerMethods in favor of inst.Peers()
- remove api handlers & just use `NewHTTPRequestMethods`
- test over cmd api & rpc

### PR DESCRIPTION
**Pull request by ramfox**
closes #1722

### CODE REVIEW

File: lib/lib.go [hunk 1]
```diff
@@ -823,6 +823,11 @@ func (inst *Instance) Dataset() DatasetMethods {
 	return DatasetMethods{inst: inst}
 }
 
+// Peer returns the PeerMethods that Instance has registered
```
Comment (dustmop) [inline lib/lib.go:826]: Please keep these *Methods functions alphabetized.

File: lib/peers.go [hunk 2]
```diff
@@ -23,8 +23,20 @@ type PeerMethods struct {
 	inst *Instance
```
Comment (dustmop) [inline lib/peers.go:23]: Change `inst *Instance` to `d dispatcher`. Now that only Dispatch is called, there's no need to keep a reference to the entire Instance.

File: api/api.go [hunk 3]
```diff
@@ -227,13 +227,12 @@ func NewServerRoutes(s Server) *mux.Router {
...
-	m.Handle(lib.AEConnect.String(), s.Middleware(ph.ConnectToPeerHandler))
-	m.Handle(lib.AEConnectAlt.String(), s.Middleware(ph.ConnectToPeerHandler))
-	m.Handle(lib.AEConnections.String(), s.Middleware(ph.ConnectionsHandler))
-	m.Handle(lib.AEConnectionsQri.String(), s.Middleware(ph.QriConnectionsHandler))
+	m.Handle(lib.AEPeers.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.list"))).Methods(http.MethodPost)
+	m.Handle(lib.AEPeer.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.info"))).Methods(http.MethodPost)
+	m.Handle(lib.AEConnect.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.connect"))).Methods(http.MethodPost)
+	m.Handle(lib.AEDisconnect.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.disconnect"))).Methods(http.MethodPost)
+	m.Handle(lib.AEConnections.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.connections"))).Methods(http.MethodPost)
+	m.Handle(lib.AEConnectedQriProfiles.String(), s.Middleware(lib.NewHTTPRequestHandler(s.Instance, "peer.connectedqriprofiles"))).Methods(http.MethodPost)
```
Comment (ramfox) [inline api/api.go:235]: using `NewHTTPRequestHandler` for each peer method!

File: api/api_test.go [hunk 4]
```diff
@@ -236,9 +236,6 @@ func TestServerReadOnlyRoutes(t *testing.T) {
 		{"PUT", "/profile/photo", 403},
 		{"POST", "/profile/poster", 403},
 		{"PUT", "/profile/poster", 403},
-		{"GET", "/peers", 403},
-		{"GET", "/peers/test_peer", 403},
-		{"GET", "/connections", 403},
```
Comment (ramfox) [inline api/api_test.go:241]: `NewHTTPRequestHandler` does not support read only & we are refactoring away from this pattern as per https://github.com/qri-io/qri/issues/1723 anyhow

File: lib/api.go [hunk 5]
```diff
@@ -54,15 +54,15 @@ const (
...
-	// AEConnectAlt initiates an explicit connection to a peer
-	AEConnectAlt = APIEndpoint("/connect/{path:.*}")
+	// AEDisconnect closes an explicit connection to a peer
+	AEDisconnect = APIEndpoint("/disconnect")
 	// AEConnections lists qri & IPFS connections
 	AEConnections = APIEndpoint("/connections")
-	// AEConnectionsQri lists qri profile connections
-	AEConnectionsQri = APIEndpoint("/connections/qri")
+	// AEConnectedQriProfiles lists qri profile connections
+	AEConnectedQriProfiles = APIEndpoint("/connections/qri")
```
Comment (ramfox) [inline lib/api.go:65]: This `APIEndpoint` var name doesn't follow the other pattern we seem to be following. The var endpoint name usually matches the lib method name, while the actual endpoint string varies.

**Review by Arqu — APPROVED** [review]
Nothing really concrete to add. Agree on the `AEConnectedQriProfiles` being a bit unfriendly. We already have a discussion I think around this and the consensus was that we have `/connections` and then we had the flavors with `/ipfs`, `/qri`... as sub routes.
The rename is good though for less mental gymnastics in the code.

**Pull request merged**